assets/app/view/game/game_log.rb

```diff
@@ -226,8 +228,36 @@ def render_date_banner(time)
         h('div.chatline', { style: { textAlign: :center } }, date)
       end
 
+      def copy_log_transcript
+        actions = @game.actions.to_h { |a| [a.id, a] }
+        logText = String.new
+        @game.log.map.group_by(&:action_id).flat_map do |action_id, entries|
+          action = actions[action_id]
+          entries.flat_map do |entry|
+            time = action&.created_at ? "[#{Time.at(action.created_at || Time.now).strftime('%R')}]" : ""
+            line = entry.message
+            if line.is_a?(Engine::Action::Message)  
+              sender = line.entity.name || line.user
+              line = "#{sender}: #{line.message}"
+            end
```

Comment:
log_text << z

Reply:
I do not know this syntax/ruby.  Is this code valid?
![image](https://user-images.githubusercontent.com/28715646/150271278-e163d2bc-49ed-4810-9843-66d9e1407996.png)
![image](https://user-images.githubusercontent.com/28715646/150271339-b5dfa036-c4ca-453b-ab4d-0bdf5384d1bd.png)

Reply:
oh poop, ok just use an array then and then finish with a join

Reply:
also, using String.new fails the linter:
![image](https://user-images.githubusercontent.com/28715646/150273796-e0403d58-99f2-4a98-ba9a-0d6ef0e05d08.png)